### DISCUSSION TITLE
Possible user error if using Deducting Times

### DISCUSSION THREAD

**Discussion by empx92** [kimai/kimai#3776] [General]
### Describe the problem

If some employees are using for example a deducting break-time and the backoffice is exporting a specific project where the deducting time is used the overall time will be false.

**For example:**
User: Doe
Customer: Kranes GmbH
Project: Hall

The user "Doe" had a Worktime from 07:00 - 17:00 with a breaktime in it of one hour, so the real worktime would be 9 hours.
If the backoffice now only exports the project "Hall" of the customer "Kranes GmbH" the worktime would be 10 hours. And if those 10 hours would be billed, the customer would pay for one hour too much.

### Describe the solution you'd like

Easiest way would be to export deducting times with a choosen value (for example "lunch break") always, even if they are not selected in export. Otherwise it may should be considered to don't use the deducting-time function or even delete it from the system.

### Describe alternatives you've considered

For the moment i managed it to check the query in the pdf export if there is a specific project-title

```
{% set pause = false %}
{% if query is defined %}
		
			{# Abfrage ob Pausen im Project enthalten sind #}
			{% for project in query.projects %}
				{% if project.name == "Pause" %}
					{% set pause = true %}
				{% else %}
					{# Inject the above project where name is "Pause" in this query 
					(but i don't really know how to manage this)#}
				{% endif %}
			{% endfor %}
{% endif %}
```

```
{% if not pause %}
	<h2 class="red center">Attention: Remember to always to export break-times too!!</h2>
{% endif %}
```

**Comment by kevinpapst** [comment]
The moment your template is rendered, everything was already selected and fetched. You cannot manipulate the query.

The easiest way is to print bold error messages if something is missing.
That's what I always do in custom templates where you have to select a customer and/or a project.

Don't try to fix the human part of the problem in the software 😁 

Deduction times are not a general "Break time" feature and have a very specific use-case and one of my clients uses it in combination with a custom export template.

Here is the repo for the bundle: https://github.com/Keleo/DeductionTimeBundle
